cmd/kraft/pkg/push/push.go

```diff
@@ -0,0 +1,192 @@
+	Registry string `long:"registry" short:"r" usage:"Registry to push to"`
+}
+
+func New() *cobra.Command {
+	cmd, err := cmdfactory.New(&Push{}, cobra.Command{
+		Short:   "Push a Unikraft unikernel package to registry",
+		Use:     "push [FLAGS] [PACKAGE]",
+		Aliases: []string{"ph"},
+		Long: heredoc.Doc(`
+			Push a Unikraft unikernel, component microlibrary to a remote location
+		`),
+		Example: heredoc.Doc(`
+			# Push the image for a project in the current working directory with tag latest
+			$ kraft pkg push -r registry.example.com
+
+			# Push the image for a project at a path with tag latest
+			$ kraft pkg push -r registry.example.com /path/to/app
+			
+			# Push the image for a project at a path with tag latest
+			$ kraft pkg push -r registry.example.com -w /path/to/app
+
+			# Push the image with a given name
+			$ kraft pkg push image-name:image-tag
+			
+			# Push the image with a given name to a specific registry
+			$ kraft pkg push -r registry.example.com image-name:image-tag`),
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "pkg",
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (opts *Push) Pre(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+	pm, err := packmanager.NewUmbrellaManager(ctx)
+	if err != nil {
+		return err
+	}
+
+	cmd.SetContext(packmanager.WithPackageManager(ctx, pm))
+
+	return nil
+}
+
+func (opts *Push) Run(cmd *cobra.Command, args []string) error {
+	var err error
+	var workdir string
+
+	if f, err := os.Stat(args[0]); err == nil && f.IsDir() {
+		workdir = args[0]
+	} else if opts.Workdir != "" {
+		workdir = opts.Workdir
+	} else if len(args) == 0 {
+		workdir, err = os.Getwd()
+		if err != nil {
+			return err
+		}
+	} else {
+		workdir = ""
+	}
+
+	if opts.Registry == "" && workdir != "" {
+		return errors.Join(err, errors.New("registry must be specified when pushing a project"))
+	}
+
+	ctx := cmd.Context()
+	norender := log.LoggerTypeFromString(config.G[config.KraftKit](ctx).Log.Type) != log.FANCY
+	ref := ""
+	if workdir != "" {
+		// Argument is a directory
+		workdir = args[0]
+
+		if opts.Registry == "" {
+			return errors.Join(err, errors.New("registry must be specified when pushing a project"))
+		}
+
+		// Read the kraft yaml specification and get the target name
+		project, err := app.NewProjectFromOptions(
+			ctx,
+			app.WithProjectWorkdir(workdir),
+			app.WithProjectDefaultKraftfiles(),
+		)
+		if err != nil {
+			return err
+		}
+
+		// Get the target name and append the 'latest' tag
+		ref = project.Name() + ":latest"
+	} else {
+		// Argument is a reference name
+		ref = args[0]
+
+		if opts.Registry != "" {
+			ref = opts.Registry + "/" + ref
+		}
+
+		// Check if a tag was specified
+		if (strings.ContainsRune(ref, ':') && strings.Split(ref, ":")[1] == "") ||
+			strings.Count(ref, ":") > 1 {
+			return errors.New("invalid tag specified")
+		}
+	}
```

Comment:
Let's remove the need to be workdir-centric. We can just look up `args[0]` via the package manager